test/e2e/components/vl-select.js

```diff
@@ -27,14 +27,54 @@ class VlSelect extends VlElement {
     return container.hasClass('is-open');
   }
 
+  async groups() {
+    if (await this.isDressed()) {
+      const selectList = await this._getSelectList();
+      const selectListItemElements = await selectList.findElements(By.css('.vl-select__group, .vl-select__item'));
+      const selectListItemTypes = await Promise.all(selectListItemElements.map((selectListItem) => selectListItem.getAttribute('class')));
+      const selectListItems = selectListItemElements.map((selectListItemElement, index) => {
+        return {
+          element: selectListItemElement,
+          isGroup: selectListItemTypes[index].indexOf('vl-select__group') !== -1,
+        };
+      });
+      return selectListItems.reduce((result, item) => {
+        if (item.isGroup) {
+          result.push(new OptionGroup(item.element, true));
+        } else {
+          result[result.length - 1].addOption(item.element);
+        }
+        return result;
+      }, []);
```

Comment:
Ik begrijp niet goed waarom de group en options samengenomen worden om dan via een boolean te declareren of het een group of option is om vervolgens op basis van de group de options te gaan opbouwen. Ik zou de volledige option select logica verplaatsen naar de group class zodat dit niet meer nodig is.

Reply:
Omdat de group headers en de opties niet in een structuur (opties onderdeel van groep) worden gerendered door choices.js, maar gewoon onder mekaar. 

`
<div class="vl-select__list" dir="ltr" role="listbox"><div class="vl-select__group" data-group="" data-id="1" data-value="Group 1" role="group">
            <div class="vl-select__heading">Group 1</div>
</div>
<div class="vl-select__item vl-select__item--choice vl-select__item--selectable is-highlighted" data-select-text="Press to select" data-choice="" data-id="2" data-value="one" data-choice-selectable="" id="choices-select-dynamic-grouping-item-choice-2" role="treeitem" aria-selected="true">
            Label one
</div>
<div class="vl-select__item vl-select__item--choice vl-select__item--selectable" data-select-text="Press to select" data-choice="" data-id="3" data-value="two" data-choice-selectable="" id="choices-select-dynamic-grouping-item-choice-3" role="treeitem">
            Label two
</div>
<div class="vl-select__group" data-group="" data-id="2" data-value="Group 2" role="group">
  <div class="vl-select__heading">Group 2</div>
</div>
...
`

Dus ik kan dat verhuizen naar de optiongroup maar dan moet ik daar in de van zijne parent alles ophalen en alles items "tussen mezelf en de volgende group" gaan ophalen ...

Reply:
Omdat ze niet gestructureerd in de HTML zitten lijkt het dus beter om deze code hier te behouden ipv ze te verhuizen naar VlSelectOptionGroup.